ms-common-auth/src/main/java/net/trajano/ms/authz/TokenResource.java

```diff
@@ -1,47 +1,35 @@
-package net.trajano.ms.example.authz;
-
-import java.io.IOException;
-import java.net.URI;
-import java.text.ParseException;
-import java.time.Instant;
-import java.time.temporal.ChronoUnit;
-import java.util.Date;
-
-import javax.annotation.security.PermitAll;
-import javax.validation.constraints.NotNull;
-import javax.ws.rs.Consumes;
-import javax.ws.rs.FormParam;
-import javax.ws.rs.HeaderParam;
-import javax.ws.rs.POST;
-import javax.ws.rs.Path;
-import javax.ws.rs.Produces;
-import javax.ws.rs.core.HttpHeaders;
-import javax.ws.rs.core.MediaType;
-import javax.ws.rs.core.UriBuilderException;
+package net.trajano.ms.authz;
 
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiParam;
+import net.trajano.ms.auth.token.ErrorCodes;
+import net.trajano.ms.auth.token.GrantTypes;
+import net.trajano.ms.auth.token.IdTokenResponse;
+import net.trajano.ms.auth.util.HttpAuthorizationHeaders;
+import net.trajano.ms.authz.internal.TokenCache;
+import net.trajano.ms.auth.spi.InternalClaimsBuilder;
+import net.trajano.ms.common.oauth.ClientValidator;
+import net.trajano.ms.common.oauth.OAuthTokenResponse;
+import net.trajano.ms.core.CryptoOps;
+import org.jose4j.jwt.JwtClaims;
+import org.jose4j.jwt.MalformedClaimException;
+import org.jose4j.jwt.NumericDate;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.stereotype.Component;
 
-import com.nimbusds.jose.JOSEException;
-import com.nimbusds.jose.JWSAlgorithm;
-import com.nimbusds.jose.jwk.source.RemoteJWKSet;
-import com.nimbusds.jose.proc.BadJOSEException;
-import com.nimbusds.jose.proc.DefaultJOSEProcessor;
-import com.nimbusds.jose.proc.JWSVerificationKeySelector;
-import com.nimbusds.jose.proc.SimpleSecurityContext;
-import com.nimbusds.jwt.JWTClaimsSet;
-
-import io.swagger.annotations.Api;
-import io.swagger.annotations.ApiParam;
-import net.trajano.ms.common.oauth.ClientValidator;
-import net.trajano.ms.common.oauth.GrantTypes;
-import net.trajano.ms.common.oauth.IdTokenResponse;
-import net.trajano.ms.common.oauth.OAuthTokenResponse;
-import net.trajano.ms.vertx.beans.TokenGenerator;
+import javax.annotation.security.PermitAll;
+import javax.ws.rs.*;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.UriBuilderException;
+import java.net.URI;
+import java.time.Duration;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'java.time.Duration'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)